### DISCUSSION TITLE
lua 脚本中生成makefile的顺序问题

### DISCUSSION THREAD

**Discussion by DanielKui** [xmake-io/xmake#3902] [Q&A]
`
includes ("./Tools/LogManager/xmake.lua")

includes ("./HXApi/HXQuoteDemo.lua")

includes ("./QuoteSvr/QuoteSvr_ST_Asio.lua")

`

比如我有一个xmake.lua中内容如上，生成makefile文件，希望里面的生成顺序也是先LogManager HXQuoteDemo  QuoteSvr_ST_Asio这样的顺序，但似乎现在的行为是随机的，如何指定顺序呢？
xmake project -k makefile --root -P .

**Comment by waruqi** [comment]
指定不了

**Reply by DanielKui** [reply]
感谢回复

如果不生成makefile ，直接使用xmake build 的方式来编译，可以保证顺序吗？
因为我想做一个自动化的脚本每天 自动编译，但工程之间是有依赖关系的先BUILD A，然后B依赖A，所以必须保证A编译完才行

**Reply by waruqi** [reply]
add_deps

**Comment by DanielKui** [comment]
> 指定不了

如果是不同的lua文件，如何指定顺序呢？add_deps只能指定在同一个lua脚本中的吧？我有A.lua  B.lua  ，B.lua中的target ("test")需要先编译A.lua中的target("testfirst")，怎么实现呢？

**Comment by waruqi** [comment]
> > 指定不了
> 
> 如果是不同的lua文件，如何指定顺序呢？add_deps只能指定在同一个lua脚本中的吧？我有A.lua B.lua ，B.lua中的target ("test")需要先编译A.lua中的target("testfirst")，怎么实现呢？

我可没说过 add_deps 只能在一个 lua 文件中用。。你哪里看来的。

**Reply by DanielKui** [reply]
哦哦，我是新手确实是不太了解
![image](https://github.com/xmake-io/xmake/assets/578118/652c625f-1b06-4f7e-804b-46e5c133811e)


我是在HXQuoteDemo.lua中的需要 先编译LogManager这个模块（定义在LogManager.lua中），是需要先includes ("LogManager")吗

**Reply by waruqi** [reply]
跟先后有啥关系，你不 includes 。xmake 怎么知道 你定义了哪些 target

**Reply by waruqi** [reply]
先多看看文档，https://xmake.io/#/zh-cn/guide/syntax_description?id=%e5%a4%9a%e7%ba%a7%e9%85%8d%e7%bd%ae

**Comment by DanielKui** [comment]
> > > 指定不了
> > 
> > 
> > 如果是不同的lua文件，如何指定顺序呢？add_deps只能指定在同一个lua脚本中的吧？我有A.lua B.lua ，B.lua中的target ("test")需要先编译A.lua中的target("testfirst")，怎么实现呢？
> 
> 我可没说过 add_deps 只能在一个 lua 文件中用。。你哪里看来的。



> 跟先后有啥关系，你不 includes 。xmake 怎么知道 你定义了哪些 target



> 跟先后有啥关系，你不 includes 。xmake 怎么知道 你定义了哪些 target

谢谢
